File: web/src/api/components/schemas/Video.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, array, boolean, type } from 'io-ts';
import { DateFromISOString } from 'io-ts-types/lib/DateFromISOString';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type Video = {
	id: Option<Int>;
	subcategoryId: Option<Int>;
	showId: Option<Int>;
	showName: Option<string>;
	showSlug: Option<string>;
	categoryName: Option<string>;
	slug: Option<string>;
	title: Option<string>;
	description: Option<string>;
	publishedAt: Option<Date>;
	durationSec: Option<Int>;
	views: Option<string>;
	videoUrl: Option<string>;
	thumbMediaId: Option<Int>;
	imageUrl: Option<string>;
	tags: Option<Array<string>>;
	published: Option<boolean>;
	createdAt: Option<Date>;
	updatedAt: Option<Date>;
};
export const VideoIO = type(
	{
		id: optionFromNullable(Int),
		subcategoryId: optionFromNullable(Int),
		showId: optionFromNullable(Int),
		showName: optionFromNullable(string),
		showSlug: optionFromNullable(string),
		categoryName: optionFromNullable(string),
		slug: optionFromNullable(string),
		title: optionFromNullable(string),
		description: optionFromNullable(string),
		publishedAt: optionFromNullable(DateFromISOString),
		durationSec: optionFromNullable(Int),
		views: optionFromNullable(string),
		videoUrl: optionFromNullable(string),
		thumbMediaId: optionFromNullable(Int),
		imageUrl: optionFromNullable(string),
		tags: optionFromNullable(array(string)),
		published: optionFromNullable(boolean),
		createdAt: optionFromNullable(DateFromISOString),
		updatedAt: optionFromNullable(DateFromISOString),
	},
	'Video',
);

File: backend/src/main/java/net/tape/model/Video.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.time.Instant;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class Video implements Serializable {
    private Long id;
    private Long subcategoryId;
    private Long showId;
    private String showName;
    private String showSlug;
    private String categoryName;
    private String slug;
    private String title;
    private String description;
    private Instant publishedAt;
    private Integer durationSec;
    private String views;
    private long viewCount;
    private String videoUrl;
    private Long thumbMediaId;
    private String imageUrl;
    private String[] tags;
    private java.util.List<Security> securities = new java.util.ArrayList<>();
    private java.util.List<VideoPerson> people = new java.util.ArrayList<>();
    private boolean published;
    private Instant createdAt;
    private Instant updatedAt;
}
